File: frontend/lib/types/serie.ts
import { Genero, Imagens } from "./conteudo";

export interface TemporadaSerie {
	id: number;
	nome: string;
	numero_temporada: number;
	quantidade_episodios: number;
	descricao: string | null;
	data_lancamento: string | null;
	capa: string | null;
}

export interface SerieListagem {
	id: number;
	titulo: string;
	titulo_original: string;
	idioma_original: string;
	descricao: string | null;
	status: string;
	data_lancamento: string | null;
	imagens: Imagens;
	generos_ids: number[];
}

export interface Serie {
	id: number;
	titulo: string;
	titulo_original: string;
	idioma_original: string;
	descricao: string | null;
	status: string;
	data_lancamento: string | null;
	imagens: Imagens;
	generos: Genero[];
	duracao_episodios: number[];
	quantidade_episodios: number;
	quantidade_temporadas: number;
	temporadas: TemporadaSerie[];
}

File: backend/schemas/serie.py
from pydantic import Field

from schemas.conteudo import ImagensConteudo

from .base import Base

class GeneroSerie(Base):
    id: int
    nome: str


class TemporadaSerie(Base):
    id: int
    nome: str
    numero_temporada: int = 0
    quantidade_episodios: int = 0
    descricao: str | None = None
    data_lancamento: str | None = None
    capa: str | None = None

class SerieListRead(Base):
    id: int
    titulo: str
    titulo_original: str
    idioma_original: str
    descricao: str | None = None
    # status: str # TMDB não disponibiliza explicitamente a lista com todos os valores possíveis para Status
    data_lancamento: str | None = None
    imagens: ImagensConteudo
    generos_ids: list[int] = Field(default_factory=list)

class SerieRead(Base):
    id: int
    titulo: str
    titulo_original: str
    idioma_original: str
    descricao: str | None = None
    status: str # TMDB não disponibiliza explicitamente a lista com todos os valores possíveis para Status
    data_lancamento: str | None = None
    imagens: ImagensConteudo
    generos: list[GeneroSerie] = Field(default_factory=list)
    duracao_episodios: list[int]= Field(default_factory=list)
    quantidade_episodios: int = 0
    quantidade_temporadas: int = 0
    temporadas: list[TemporadaSerie] = Field(default_factory=list)